castle/cms/upgrades/upgrade_2_6_18.py

```diff
@@ -0,0 +1,25 @@
+from plone.registry import field
+from plone.registry import Record
+from plone.registry.interfaces import IRegistry
+from plone import api
+from Products.CMFCore.utils import getToolByName
+from zope.component import getUtility
+
+PROFILE_ID = 'profile-castle.cms:2_6_18'
+
+
+def upgrade(context, logger=None):
```

Comment:
You'll want to add a reference to the castle/cms/upgrades.zcml config, and then create an upgrade profile in castle/cms/profiles/ -- and creating the upgrade profile might mean that some of this code wouldn't be necessary.

That's the way we typically add these kinds of upgrades so far.